src/lib.rs

```diff
@@ -7,47 +7,15 @@
 #![cfg_attr(not(feature = "unstable"), allow(warnings))]
 
 extern crate deque;
-#[macro_use]
-extern crate lazy_static;
 #[cfg(feature = "unstable")]
 extern crate futures;
 extern crate libc;
 extern crate num_cpus;
 extern crate rand;
+extern crate rayon_core;
 
-#[macro_use]
-mod log;
-
-mod configuration;
-mod latch;
-mod join;
-mod job;
 pub mod par_iter;
 pub mod prelude;
-#[cfg(test)]
 mod test;
-mod registry;
-#[cfg(feature = "unstable")]
-mod future;
-mod scope;
-mod sleep;
-#[cfg(feature = "unstable")]
-mod spawn_async;
-mod thread_pool;
-mod unwind;
-mod util;
 
-pub use configuration::Configuration;
-pub use configuration::PanicHandler;
-pub use configuration::InitError;
-pub use configuration::dump_stats;
-pub use configuration::initialize;
-pub use thread_pool::ThreadPool;
-pub use join::join;
-pub use scope::{scope, Scope};
-#[cfg(feature = "unstable")]
-pub use spawn_async::spawn_async;
-#[cfg(feature = "unstable")]
-pub use spawn_async::spawn_future_async;
-#[cfg(feature = "unstable")]
-pub use future::RayonFuture;
+pub use rayon_core::*;
```

Comment:
I'd prefer explicit re-exports for two reasons:

1. It makes the API surface of a particular `rayon` version consistent, rather than depending on what `rayon-core` lies beneath.

2. It works better with rustdoc, which only writes globs as a single re-exports link, whereas direct `pub use rayon_core::foo` creates inline documentation.